<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <!-- Robot name parameter -->
    <arg name="robot_name" default="locobot" />
    
    <!-- Configuration file path -->
    <arg name="poses_config" default="$(find object_scout)/config/poses.yaml" />
    
    <!-- Visualization parameters -->
    <arg name="rviz_frame" default="map" />
    <arg name="rvizconfig" default="$(find yolo_vision)/rviz/xslocobot_description_yolo.rviz" />
    <arg name="use_rviz" default="true" doc="Set to false to disable RViz visualization" />
    
    <!-- Launch the test navigation node -->
    <node name="test_navigation" pkg="object_scout" type="test_navigation_node.py" output="screen">
        <param name="robot_name" value="$(arg robot_name)" />
        <param name="poses_config" value="$(arg poses_config)" />
    </node>
    
    <!-- RViz (conditionally launched) -->
    <group if="$(arg use_rviz)">
        <node name="rviz" pkg="rviz" type="rviz" 
                args="-f $(arg rviz_frame) -d $(arg rvizconfig)"
                ns="$(arg robot_name)">
            <remap from="/clicked_point" to="clicked_point"/>
            <remap from="/initialpose" to="initialpose"/>
            <remap from="/move_base_simple/goal" to="move_base_simple/goal"/>
        </node>
    </group>
</launch>